server/controllers/profile.js

```diff
@@ -38,4 +36,36 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
       profile: profile,
     },
   });
-});
\ No newline at end of file
+});
+
+// @route GET /profile
+// @desc Get list of pet sitter profiles by location or get all
+// @access Public
+exports.searchProfiles = asyncHandler(async (req, res, next) => {
+
+  if (req.query.location) {
+
+    const profiles = await Profile.find({ accountType: 'pet_sitter', address: { $regex: req.query.location, $options: "i" }, pay: { $ne: null } });
+
+    res.status(200).json({
+      profiles: profiles
+    });
+
+  } else if (req.query.location === '') {
+
+    const profiles = await Profile.find({ accountType: 'pet_sitter', address: { $ne: null | '' }, pay: { $ne: null } });
```

Comment:
Both queries are almost identical, why don't we combine them? We can store the query filter in a variable based on whether query.location exists or not